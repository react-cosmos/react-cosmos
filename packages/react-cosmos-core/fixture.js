module.exports = require('./dist/fixture/index.js');
