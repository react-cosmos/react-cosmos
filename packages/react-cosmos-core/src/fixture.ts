export * from './fixture/useSelect/index.js';
export * from './fixture/useValue/index.js';
export * from './fixture/Viewport.js';
