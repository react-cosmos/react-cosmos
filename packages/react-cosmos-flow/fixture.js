// @flow

import type { Node, ComponentType, ElementRef } from 'react';

export type FixtureType<P: {}> = {
  component: ComponentType<P>,
  name?: string,
  namespace?: string,
  props?: P,
  children?: Node, // Deprecated in favor of props.children
  init?: ({ compRef: ?ElementRef<ComponentType<P>> }) => Promise<any>
};

// Deprecated. Use this instead:
//  import { createFixture } from 'react-cosmos'
export function createFixture<P: {}, F: FixtureType<P>>(fixture: F) {
  return fixture;
}
