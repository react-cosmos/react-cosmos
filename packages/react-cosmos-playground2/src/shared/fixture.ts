export function removeFixtureNameExtension(fixtureName: string) {
  return fixtureName.replace(/\.(j|t)sx?$/, '');
}

export function removeFixtureNameSuffix(
  fixtureNameWithoutExtension: string,
  suffix: string
) {
  return fixtureNameWithoutExtension.replace(new RegExp(`\\.${suffix}$`), '');
}
