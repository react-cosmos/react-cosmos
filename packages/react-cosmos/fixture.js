require('regenerator-runtime/runtime');
module.exports = require('./dist/fixture');
