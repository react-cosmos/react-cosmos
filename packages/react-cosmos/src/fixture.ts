import 'regenerator-runtime/runtime';
export { Viewport, useValue } from 'react-cosmos-shared2/FixtureLoader';
