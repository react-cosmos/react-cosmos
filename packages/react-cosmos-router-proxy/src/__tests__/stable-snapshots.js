import createTestContext from 'react-cosmos-test/enzyme';
import fixture from '../__fixtures__/stableKeys';
import createRouterProxy from '../';
import toJson from 'enzyme-to-json';

const proxies = [createRouterProxy()];

const { mount, getWrapper } = createTestContext({ fixture, proxies });

test('props are stable', () => {
  mount();
  const componentOneLocationPropKey = getWrapper()
    .find('MyComponent')
    .props().location.key;

  mount();
  const componentTwoLocationPropKey = getWrapper()
    .find('MyComponent')
    .props().location.key;

  expect(componentOneLocationPropKey).toEqual(componentTwoLocationPropKey);
});

test('snapshot is stable', () => {
  mount();
  expect(toJson(getWrapper())).toMatchSnapshot();
});
