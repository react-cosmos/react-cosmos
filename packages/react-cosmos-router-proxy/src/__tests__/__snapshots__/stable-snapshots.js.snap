// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot is stable 1`] = `
<withRouter(MyComponent)>
  <Route>
    <MyComponent
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "mocked",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "mocked",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
      location={
        Object {
          "hash": "",
          "key": "mocked",
          "pathname": "/",
          "search": "",
          "state": undefined,
        }
      }
      match={
        Object {
          "isExact": true,
          "params": Object {},
          "path": "/",
          "url": "/",
        }
      }
    >
      <div>
        MyComponent
      </div>
    </MyComponent>
  </Route>
</withRouter(MyComponent)>
`;
